pkg/operators/ustack/ustack.go

```diff
@@ -0,0 +1,247 @@
+	}
+	err := instance.init(gadgetCtx)
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
+
+func (o *Operator) Priority() int {
+	return Priority
+}
+
+type OperatorInstance struct {
+	userStackMap *ebpf.Map
+	symbolizer   *symbolizer.Symbolizer
+
+	subscriptions map[datasource.DataSource][]func(ds datasource.DataSource, data datasource.Data) error
+}
+
+func (o *OperatorInstance) init(gadgetCtx operators.GadgetContext) error {
+	logger := gadgetCtx.Logger()
+
+	for _, ds := range gadgetCtx.GetDataSources() {
+		for _, in := range ds.GetFieldsWithTag("type:" + ebpftypes.UserStackTypeName) {
+			if in == nil {
+				continue
+			}
+			in.SetHidden(true, false)
+
+			// Optional fields: don't error out if nil
+			containerPidField := ds.GetField("runtime.containerPid")
+			containerNameField := ds.GetField("runtime.containerName")
+			commField := ds.GetField("proc.comm")
+
+			pidField := ds.GetField("proc.pid")
+			if pidField == nil {
+				gadgetCtx.Logger().Warnf("no pid field found")
+				continue
+			}
+
+			stackField := in.GetSubFieldsWithTag("name:stack_id")
+			if len(stackField) != 1 {
+				gadgetCtx.Logger().Warnf("no stack_id field found")
+				continue
+			}
+
+			inodeField := in.GetSubFieldsWithTag("name:exe_inode")
+			if len(inodeField) != 1 {
+				gadgetCtx.Logger().Warnf("no inode field found")
+				continue
+			}
+			mtimeSecField := in.GetSubFieldsWithTag("name:mtime_sec")
+			if len(mtimeSecField) != 1 {
+				gadgetCtx.Logger().Warnf("no mtime_sec field found")
+				continue
+			}
+			mtimeNsecField := in.GetSubFieldsWithTag("name:mtime_nsec")
+			if len(mtimeNsecField) != 1 {
+				gadgetCtx.Logger().Warnf("no mtime_nsec field found")
+				continue
+			}
+
+			targetName, err := annotations.GetTargetNameFromAnnotation(logger, "ustack", in, userStackTargetNameAnnotation)
+			if err != nil {
+				logger.Warnf("Failed to get target name for enum field %q: %v", in.Name(), err)
+				continue
+			}
+			out, err := ds.AddField(targetName, api.Kind_String, datasource.WithSameParentAs(in))
+			if err != nil {
+				return err
+			}
+			converter := func(ds datasource.DataSource, data datasource.Data) error {
+				outString := ""
+
+				inBytes := stackField[0].Get(data)
+				stackId := ds.ByteOrder().Uint32(inBytes)
+				pid, _ := pidField.Uint32(data)
+				containerPid := uint32(0)
+				if containerPidField != nil {
+					containerPid, _ = containerPidField.Uint32(data)
+				}
+				containerName := ""
+				if containerNameField != nil {
+					containerName, _ = containerNameField.String(data)
+				}
+				comm := ""
+				if commField != nil {
+					comm, _ = commField.String(data)
+				}
+				inode, _ := inodeField[0].Uint64(data)
+				mtimeSec, _ := mtimeSecField[0].Uint64(data)
+				mtimeNsec, _ := mtimeNsecField[0].Uint32(data)
+
+				// If user stacks are disabled
+				if inode == 0 {
+					return nil
+				}
```

Comment:
Should it be the first operation of avoid unnecessary work if the user stacks are disabled?